warp10/src/main/sh/warp10.sh

```diff
@@ -17,506 +17,372 @@
-      [[ $(getent group ${WARP10_GROUP}) ]] || addgroup ${WARP10_GROUP}
-      adduser --system --home ${WARP10_HOME} --no-create-home --shell /bin/bash --ingroup ${WARP10_GROUP} ${WARP10_USER}
-    elif [[ $(command -v dscl || true) ]]; then
-      if [[ ! $(grep -q "^${WARP10_GROUP}:" /etc/group || true) ]]; then
-        dscl . -create /Groups/${WARP10_GROUP}
-        dscl . -create /Groups/${WARP10_GROUP} gid 10042
-      fi
-      gid=$(dscl . -read /Groups/${WARP10_GROUP} | awk '($1 == "PrimaryGroupID:") { print $2 }')
-      dscl . -create /Users/${WARP10_USER}
-      dscl . -create /Users/${WARP10_USER} UniqueID 10042
-      dscl . -create /Users/${WARP10_USER} PrimaryGroupID ${gid}
-      dscl . -create /Users/${WARP10_USER} UserShell /bin/bash
-    else
-      echo "Cannot create the ${WARP10_USER} user with home directory ${WARP10_HOME}. Create it manually then run the script again."
-      exit 1
-    fi
+    die "ERROR: ${WARP10_USER} user does not exist, please create it before running this script."
   fi
 
-  #
-  # If config file already exists then.. exit
-  #
+  ##
+  ##
+  ## Exit if config files already exist
+  ##
   getConfigFiles
-  if [[ ! -z "${CONFIG_FILES}" ]]; then
-    echo "Config file already exists - Abort bootstrap..."
-    exit 2
+  if [ -n "${CONFIG_FILES:-}" ]; then
+    die "ERROR: Configuration files already exist - Abort initialization."
   fi
 
-  # Fix ownership
-  echo "Fix ownership.."
-  echo "WARP10_HOME: ${WARP10_HOME}"
-
-  # Trailing slash is needed when ${WARP10_HOME} is a symlink
-  chown -R ${WARP10_USER}:${WARP10_GROUP} ${WARP10_HOME}/
-
   # Fix permissions
-  echo "Fix permissions.."
-  chmod 750 ${WARP10_HOME}
-  chmod 755 ${WARP10_HOME}/bin
-  chmod 755 ${WARP10_HOME}/etc
-  chmod 755 ${WARP10_HOME}/logs
-  chmod 755 ${WARP10_HOME}/macros
-  chmod 755 ${WARP10_HOME}/jars
-  chmod 755 ${WARP10_HOME}/lib
-  chmod 755 ${WARP10_HOME}/templates
-  chmod 755 ${WARP10_HOME}/runners
-  chmod 755 ${WARP10_HOME}/etc/throttle
-  chmod 755 ${WARP10_HOME}/etc/trl
-  chmod 755 ${WARP10_HOME}/etc/bootstrap
-  chmod 644 ${WARP10_HOME}/etc/bootstrap/*.mc2
-  chmod 755 ${WARP10_HOME}/bin/*.sh
-  chmod 755 ${WARP10_HOME}/bin/*.init
-  chmod 644 ${WARP10_HOME}/bin/*.service
-  chmod 644 ${WARP10_HOME}/bin/warp10-@VERSION@.jar
-  chmod -R 755 ${WARP10_HOME}/datalog
-  chmod -R 755 ${WARP10_HOME}/datalog_done
-  chmod -R 755 ${WARP10_HOME}/leveldb
-
-  #
-  # Test access to WARP10_HOME for WARP10_USER
-  #
-  su ${WARP10_USER} -c "ls ${WARP10_HOME} >/dev/null 2>&1"
-  if [ $? != 0 ]; then
-    echo "ERROR: ${WARP10_USER} user cannot access to ${WARP10_HOME}"
-    exit 1
-  fi
-
-  #
-  # ${WARP10_HOME} != ${WARP10_DATA_DIR}
-  # A dedicated data directory has been provided
-  # Move data to ${WARP10_DATA_DIR}/etc, ${WARP10_DATA_DIR}/logs, ${WARP10_DATA_DIR}/leveldb..
-  #
-  if [ "${WARP10_DATA_DIR}" != "${WARP10_HOME}" ]; then
-    echo "WARP10_DATA_DIR: ${WARP10_DATA_DIR}"
+  echo "Fix permissions"
```

Comment:
permissions should already be correct since they are in the tar archive. Again this is going beyond what the script should do.

Reply:
done